Tests/ExtDataRoot_GridComp.F90

```diff
@@ -556,8 +556,8 @@ subroutine CompareState(State1,State2,tol,rc)
             call ESMF_StateGet(State2,trim(nameList(i)),field2,_RC)
             call ESMF_FieldGet(field1,rank=rank1,_RC)
             call ESMF_FieldGet(field2,rank=rank2,_RC)
-            all_undef1 = is_field_undef(field1,_RC)
-            all_undef2 = is_field_undef(field2,_RC)
+            all_undef1 = FieldIsConstant(field1,MAPL_UNDEF,_RC)
+            all_undef2 = FieldIsConstant(field2,MAPL_UNDEF,_RC)
```

Comment:
Note for future extension of this layer.     We could overload `==` and re-express this as:
```
all_undef = all (field1 == MAPL_UNDEF)
```